spec-0007/index.md

```diff
@@ -0,0 +1,183 @@
+(2) they avoid using global state (from `np.random.mtrand._rand`).
+
+Our recommendation here is a deprecation strategy which does not in _all_ cases adhere to the Hinsen[^hinsen] principle.
+
+The [deprecation strategy](https://github.com/scientific-python/specs/pull/180#issuecomment-1515248009) is:
+
+1. Accept both `rng` and `random_state` keyword arguments.
+2. If `rng=None`, handle `random_state` as in legacy behavior (see above), except use a compatible Generator instead of RandomState.
+   A DeprecationWarning is raised to warn about a future change in behavior.
+3. After <X time>, use only `rng`, seeding with `default_rng(rng)`.
+   Raise an error if `random_state` is provided.
+4. At a time of the library's choosing, remove any machinery related to `random_state`.
+
+### Impact
+
+The following users will be affected:
+
+1. Those who use `np.random.seed`. The proposal will do away with that global seeding mechanism, meaning that code that relies on it will, after a certain deprecation period, start seeing a different stream of random numbers than before. To ensure that this does not go unnoticed, the library should raise a `FutureWarning` if `np.random.seed` was called earlier (we show how to do that further down).
+
+   Such code will, in effect, go from being seeded to being unseeded.
+   To avoid that from happening, the code will have to be modified to pass in explicitly an `rng` argument on each function call.
+
+2. Those who do not seed. Their code will, after the deprecation period, use the newly proposed default. Since they were already not requesting repeatable sequences, and since the underlying _distributions_ of pseudo-random numbers did not change, they should be unaffected.
+
+3. Users of `random_state=...`. Support for the `random_state` argument may be dropped eventually, but meanwhile we can provide clear guidance, via deprecation warnings and documentation, on how to migrate to the new `rng` keyword.
+
+[^hinsen]: The Hinsen principle states, loosely, that code should, whether executed now or in the future, return the same result, or raise an error.
+
+### Code
+
+As an example, consider how SciPy would transition from the `seed` to the `rng` keyword using a decorator.
+This is implemented using:
+1. A `check_random_state` function which normalizes either old (`seed`) or new (`rng`) input to a `Generator` object.
+   If neither `seed` nor `rng` was passed but the user has previously called `np.random.seed()`
+   this function gives a `FutureWarning` because the behavior will change as noted in
+   the Impact section point 1.
+2. A decorator to deal with the `seed` to `rng` keyword rename.  In future versions, this will deprecate the `seed` keyword. Meanwhile, it ensures that the documentation and auto-completion only advertises the new parameter name.
```

Comment:
```suggestion
2. A decorator to deal with the `seed` to `rng` keyword rename.  In future versions, this will deprecate the keyword-only `seed` parameter. Meanwhile, it ensures that the documentation and auto-completion only advertises the new parameter name.
```

And maybe a note that this approach is only correct if the deprecated parameter is keyword-only?